Update faq.md (#780)
returntocorp/semgrep-docs: docs/faq.md

- Semgrep operates directly on source code, whereas CodeQL requires a buildable environment
- Semgrep is LGPL-2.1 and free to run anywhere; CodeQL is not open source and you must pay to run it on any non-open-source code
- Semgrep supports autofixes; CodeQL does not.

> - Semgrep focuses on speed and ease of use. Semgrep doesn’t require a buildable environment, it doesn’t yet provide some of the analysis features of CodeQL. Semgrep has [limited intraprocedural dataflow](/writing-rules/data-flow/data-flow-overview/) and the proprietary [DeepSemgrep](/deepsemgrep/) extension to Semgrep provides parity in terms of interprocedural dataflow analysis for a subset of supported languages.

Reviewer: @ievans link fixed. Thanks for your contribution!

Just asking some questions to improve my product knowledge and this update:
- How does the fact that Semgrep doesn't require a buildable environment correspond to the information that Semgrep does not yet provide some analysis features of CodeQL? Is this information related really?
- Are we trying to say that: what Semgrep misses compared to CodeQL DeepSemgrep manages to do?
- `Semgrep provides parity` parity with what? With CodeQL? If we are talking about parity as equality, we need to say what we are comparing it to in one sentence.

Thank you again and looking forward to your answers, I was not sure about this text even in its previous form...

Reviewer: ```suggestion
- Semgrep focuses on speed and ease of use. Semgrep doesn’t require compiled code, it doesn’t yet provide some of the analysis features of CodeQL. Semgrep has [limited intraprocedural dataflow](/writing-rules/data-flow/data-flow-overview/) and the proprietary [DeepSemgrep](/deepsemgrep/) extension to Semgrep provides similar capabilities as CodeQL in terms of interprocedural dataflow analysis for a subset of supported languages.
```

> - Semgrep focuses on speed and ease of use. Because it doesn’t require a buildable environment, it doesn’t yet have some of the analysis features that CodeQL does. Semgrep has [limited intraproceedural dataflow](/writing-rules/data-flow/data-flow-overview/) and the proprietary [DeepSemgrep](./deepsemgrep/) extension to Semgrep provides parity in terms of interprocedural dataflow analysis for a subset of supported languages.

Reviewer: ```suggestion
- Semgrep focuses on speed and ease of use. Because Semgrep doesn’t require a buildable environment, it doesn’t yet provide some of the analysis features of CodeQL. Semgrep has [limited intraprocedural dataflow](/writing-rules/data-flow/data-flow-overview/) and the proprietary [DeepSemgrep](/deepsemgrep/) extension to Semgrep provides parity in terms of interprocedural dataflow analysis for a subset of supported languages.
```
